app/routerTransition.js

```diff
@@ -23,141 +24,531 @@ import notify from "actions/NotificationActions";
+            ) !== -1;
+
+        this._initConnectionManager(urls);
+
+        if (!latenciesEstablished) {
+            console.log("doLatencyUpdate initial");
+            this.doLatencyUpdate(true)
+                .then(
+                    this._initiateConnection.bind(
+                        this,
+                        nextState,
+                        replaceState,
+                        callback,
+                        appInit
+                    )
+                )
+                .catch(() => {
+                    console.log("catch doLatency");
+                });
+        } else {
+            this._initiateConnection(
+                nextState,
+                replaceState,
+                callback,
+                appInit
+            );
+        }
+    }
+
+    /**
+     * Updates the latency of all target nodes
+     *
+     * @param refresh boolean true reping all existing nodes
+     *                        false only reping all reachable nodes
+     * @returns {Promise}
+     */
+    doLatencyUpdate(refresh = true) {
+        return new Promise((resolve, reject) => {
+            console.log("doLatencyUpdate begin");
+            // if for some reason this method is called before connections are setup via willTransitionTo,
+            // initialize the manager
+            if (this._connectionManager == null) {
+                this._initConnectionManager();
+            }
+            if (refresh) {
+                console.log("doLatencyUpdate refresh");
+                this._connectionManager.urls = this._getNodesToConnectTo(true);
+            }
+            console.log(SettingsStore.getState().apiLatencies);
+            this._connectionManager
+                .checkConnections()
+                .then(res => {
+                    console.log("doLatencyUpdate ping done");
+                    // resort the api nodes with the new pings
+                    this._connectionManager.urls = this._getNodesToConnectTo(
+                        false,
+                        res
+                    );
+                    // update the latencies object
+                    SettingsActions.updateLatencies(res);
+                    resolve();
+                })
+                .catch(err => {
+                    console.log("doLatencyUpdate error");
+                    console.log(err);
+                    reject(err);
+                });
+        });
+    }
+
+    _initConnectionManager(urls = null) {
+        if (urls == null) {
+            console.log("_initConnectionManager");
+            urls = this._getNodesToConnectTo(true);
+        }
+        // decide where to connect to first
+        let connectionString = this._getFirstToTry(urls);
+
+        this._connectionManager = new Manager({
+            url: connectionString,
+            urls: []
+        });
+        // don't give urls in constructor of Manager, it removes the entry given with url.
```

Comment:
There's a reason it removes the url entry, andif you set it the way you do you'll probably break the fallback logic. The ConnectionManager constructor simply filters out the url entry, which does not change the sort order.

Reply:
In the current solution, and to my understanding of the Manager class, it does only try to reconnect to the same node while falling back, which I think is reasonable in case of mere network interruption. Does anything else happen in the manager?

The same logic was also implemented in the old logic https://github.com/blockchainprojects/bitshares-ui/blob/7a2420f858ed83a1e26bee734f0f7dee0b25af49/app/routerTransition.js#L260 when the nodes got repinged, that's where I had the idea from.